dotnet-tools.json

```diff
@@ -15,10 +15,10 @@
       ]
     },
     "boogie": {
-      "version": "2.16.0",
+      "version": "3.0.1",
```

Comment:
(blocking) Should be 3.0.0 to match the actual dependency: https://github.com/dafny-lang/dafny/blob/master/Source/DafnyCore/DafnyCore.csproj#L35

Ideally we'd at least have a test that asserts they are the same somehow, if not an actual way of keeping them in sync automatically.

Reply:
Oh, oops. I was using 3.0.1 in my proof dependency PR and forgot it wasn't merged yet. :)

Reply:
I'll see if I can construct a test that compares the versions without too much effort.